core/src/main/java/org/apache/iceberg/actions/SizeBasedFileRewriter.java

```diff
@@ -101,8 +101,6 @@ public abstract class SizeBasedFileRewriter<T extends ContentScanTask<F>, F exte
 
   public static final long MAX_FILE_GROUP_SIZE_BYTES_DEFAULT = 100L * 1024 * 1024 * 1024; // 100 GB
 
-  private static final long SPLIT_OVERHEAD = 5 * 1024;
```

Comment:
This split overhead did very little given that it is only 5 KB. Row group size discrepancies are usually a few megabytes. I will have a follow-up PR to make our split planning less sensitive.